src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -138,7 +138,28 @@ public void On(string methodName, Type[] parameterTypes, Func<object[], Task> ha
 
         private async Task<ReadableChannel<object>> StreamAsyncCore(string methodName, Type returnType, object[] args, CancellationToken cancellationToken)
         {
-            var irq = InvocationRequest.Stream(cancellationToken, returnType, GetNextId(), _loggerFactory, out var channel);
+            var irq = InvocationRequest.Stream(CancellationToken.None, returnType, GetNextId(), _loggerFactory, out var channel);
+
+            if (cancellationToken.CanBeCanceled)
```

Comment:
After a bit more thinking I think this is wrong for streaming. We need to cancel the stream itself, not the invocation to start streaming. It’s weird to have the token affect both the initial call and subsequent calls via the channel